src/helpers/workspace.ts

```diff
@@ -17,6 +25,49 @@ export function getWorkspaceRoot(ignoreException: boolean = false): string | und
   return workspaceRoot;
 }
 
+export async function getWorkspaces(): Promise<TruffleWorkspace[]> {
+  const workspaces: TruffleWorkspace[] = [];
+
+  await Promise.all(
+    workspace.workspaceFolders!.map(async (ws) => {
+      workspaces.push(...(await getWorkspaceFiles(ws.uri.fsPath)));
+    })
+  );
+
+  if (workspaces === undefined) {
+    const error = new Error(Constants.errorMessageStrings.VariableShouldBeDefined("Workspace root"));
+    Telemetry.sendException(error);
+    throw error;
+  }
+
+  return workspaces;
+}
+
 export function isWorkspaceOpen(): boolean {
   return !!(workspace.workspaceFolders && workspace.workspaceFolders[0].uri.fsPath);
 }
+
+async function getWorkspaceFiles(dirPath: string, truffleWorkSpaces?: TruffleWorkspace[]): Promise<TruffleWorkspace[]> {
+  const files = fs.readdirSync(dirPath);
+
+  truffleWorkSpaces = truffleWorkSpaces || [];
+
+  await Promise.all(
+    files.map(async (file) => {
+      if (file.includes("node_modules")) return;
+
+      if (fs.statSync(`${dirPath}/${file}`).isDirectory()) {
+        truffleWorkSpaces = await getWorkspaceFiles(`${dirPath}/${file}`, truffleWorkSpaces);
+      } else {
+        if (file === Constants.defaultTruffleConfigFileName)
+          truffleWorkSpaces!.push({
+            dirName: path.dirname(`${dirPath}/${file}`).split(path.sep).pop()!.toString(),
+            workspace: Uri.parse(dirPath),
+            truffleConfig: Uri.parse(`${dirPath}/${file}`),
+          });
+      }
+    })
+  );
+
+  return truffleWorkSpaces;
```

Comment:
I'm not a huge fan of mutating parameters then returning them back, its just a habit to make a function idempotent to make your return variable inside the function and treat all inputs as `readonly`

Reply:
I've realized Glob is already installed in the project. So, I revamped all this function and now it's more clear and with no mutations...